sdk/servicebus/azservicebus/message.go

```diff
@@ -1,12 +1,26 @@
 package azservicebus
 
-import "github.com/Azure/azure-sdk-for-go/sdk/servicebus/azservicebus/internal"
+import (
+	"github.com/Azure/azure-amqp-common-go/v3/uuid"
+	"github.com/Azure/azure-sdk-for-go/sdk/servicebus/azservicebus/internal"
+	"github.com/Azure/go-amqp"
+)
+
+const (
+	annotationPartitionKey    = "x-opt-partition-key"
+	annotationViaPartitionKey = "x-opt-via-partition-key"
+)
 
 // Message represents a sendable message, including only user-settable properties.
 // A message that has been received from Service Bus will be of type `ReceivedMessage`.
 type Message struct {
-	Body []byte
-	ID   string
+	Body                    []byte
+	ID                      string
+	PartitionKey            *string
```

Comment:
Does this and `TransactionPartitionKey` need to be pointer-to-type?

Reply:
nil is actually valid in partitionKey case.

I can't do transactions quite yet so I can't test the other but it follows they'd use the same rules.